rust/main/hyperlane-base/src/cache/moka/hyperlane_cache.rs

```diff
@@ -0,0 +1,95 @@
+use super::{BaseCache, CacheResult, Expiration, ExpirationType};
+use crate::cache::{CacheError, HyperlaneCache};
+use async_trait::async_trait;
+use hyperlane_core::H256;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use std::fmt::Debug;
+
+/// Cache for storing function calls with serializable results
+#[derive(Debug, Clone)]
+pub struct HyperlaneMokaCache(BaseCache);
+
+/// Used when the function has no parameters
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NoParams;
```

Comment:
I wonder if `NoParams` is actually part of interface, not part of concrete implementation, and should be next to `HyperlaneCache`?

Reply:
As discussed, even though the interface does not rely on `NoParams`, we should still move it there so it would be a suggested parameter to use when storing/retrieving cached data.

[Fix](https://github.com/Mantas-M/hyperlane-monorepo/commit/f9dc61b63813d69a017ca628d592b2d2b73aab67)